feat(storybook): upgrade from v6 to v10

lsagetlethias opened the pull request:
## Contexte
Migration de la configuration Storybook de Webpack v6 vers v10 Vite, plus quelques nettoyages et adaptations.

## Ajout / suppression de dépendances
- Ajout : `@storybook/react-vite`, `@storybook/addon-docs`, `@vueless/storybook-dark-mode` (et dépendances liées à Storybook v10 / Vite).
- Suppression : `storybook-dark-mode`.

## Modifications Storybook
- Migration de la config Storybook Webpack → Vite + passage en TS
- Adaptation dark mode / theming :
  - utilisation de `@vueless/storybook-dark-mode`
  - passage à l'utilisation du prop `theme` sur le container plutôt que d'overrider `context`
- Ajout de `@storybook/addon-docs` dans la config

## Petits fixs additionnels hors Storybook
- `.eslintignore` : autorise le dossier `.storybook` (`!.storybook`)
- Ajustements de formatage/imports et typage
- Stories : titres explicites (ex. `components/...`)
- Composants : ajouts des props manquantes obligatoires

lsagetlethias commented on stories/getStory.tsx:
```diff
@@ -46,14 +43,21 @@ export function getStoryFactory<Props extends Record<string, any>>(params: {
+        isFirstStory,
+        lang,
+        ...props
+    }) => {
```
wrong

lsagetlethias commented on stories/intro.mdx:
```diff
@@ -0,0 +1,41 @@
+import { Meta } from "@storybook/addon-docs/blocks";
```
wrong

lsagetlethias commented on .storybook/preview.tsx:
```diff
@@ -0,0 +1,132 @@
+                    }
+                }
+            }
+        },
```
then why the typing is ok?

garronej commented:
This is such a beautiful Christmas present!  
Thank you very much @lsagetlethias, I’m very excited.

I’ve been delighted to see how recent improvements in LLM capabilities, especially Codex, now make it possible to vibe-code react-dsfr / MUI / tss-react UIs in a very convincing way.

This feels like a bit of a game changer. I was able to vibe-code this in about an hour:
https://vite-insee-starter.demo-domain.ovh/

Your PR, by enabling MCP, will further facilitate this approach. I’d be really excited to get your feedback on how well it works for you and how you set it up. It might even be worth adding a documentation page about it.

It might also be time to revisit the DSFR theme for https://nextra.site/ that I started a while ago  
(https://github.com/codegouvfr/nextra-theme-dsfr) but never finished, mainly because I was discouraged by the amount of repetitive and tedious work involved.  
Now, AI might be able to one-shot it.

It feels like it’s about time we have a modern solution to generate extensible DSFR websites from a Markdown corpus.

I’ll test your PR now, make sure everything works, and if it does, I’ll merge it.

lsagetlethias commented:
I'll take a look, the changes I made were only for the upgrade so we alt libs there's maybe some tweaks to do.

lsagetlethias commented:
@garronej all addressed issues are fixed now.
But:
> Stories that previously pointed to different anchors in the doc view now all open the canvas instead.

I don't understand what you're talking about 🤔

lsagetlethias commented on .storybook/customTheme.ts:
```diff
@@ -0,0 +1,35 @@
+import { create } from "storybook/theming";
```
wrong: https://storybook.js.org/docs/configure/user-interface/theming

lsagetlethias commented on .storybook/main.ts:
```diff
@@ -0,0 +1,26 @@
+    ],
+    addons: [
+        "@vueless/storybook-dark-mode",
+        "@storybook/addon-links",
```
Wrong: https://storybook.js.org/docs/addons/addon-migration-guide#dependency-updates

lsagetlethias commented:
There was so many undocumented changes... Always had to tinker and search inside ts def files, or even compiled js files to find what the purpose of some params 🥲

garronej commented:
Love it, thanks a lot!
